### PR TITLE
1589509715 Workspace Add Teams

### PR DESCRIPTION
**Pull request by aswath-m**

### CODE REVIEW

File: moncli/api_v2/handlers.py [hunk 1]
```diff
@@ -2191,4 +2192,234 @@ def create_workspace(name: str, kind: WorkspaceKind, *args, **kwargs):
...
+        Parameters
+
+            workspace_id: str
+                The workspace's unique identifier.
+            team_ids: list()
+                Team ids to subscribe to the workspace.
+        
+        Returns
+
+            data : `dict`
```
Comment (Ashatz) [inline moncli/api_v2/handlers.py:2373]: Please use the return fields from the Team object.  See _get_teams_ and copy.

File: moncli/api_v2/handlers.py [hunk 2]
```diff
@@ -2191,4 +2192,234 @@ def create_workspace(name: str, kind: WorkspaceKind, *args, **kwargs):
...
+                A monday.com workspace in dictionary form.
+        
+        Return Fields
+
+            id: str
```
Comment (Ashatz) [inline moncli/api_v2/handlers.py:2412]: Please use the return fields from the Team object. See get_teams and copy.

File: moncli/entities/client.py [hunk 3]
```diff
@@ -1273,7 +1281,337 @@ def create_workspace(self, name: str, kind: WorkspaceKind, *args, **kwargs):
...
+                Team ids to subscribe to the workspace.
+        
+        Returns
+
+            data : `dict`
+                A monday.com workspace in dictionary form.
+        
+        Return Fields
+
+            account : `moncli.entities.Account`
```
Comment (Ashatz) [inline moncli/entities/client.py:1463]: These are the return fields for the User object. Please use the return fields from the Team object. See get_teams and copy.

File: moncli/entities/client.py [hunk 4]
```diff
@@ -1273,7 +1281,337 @@ def create_workspace(self, name: str, kind: WorkspaceKind, *args, **kwargs):
...
+        Parameters
 
+            workspace_id: str
+                The workspace's unique identifier.
+            team_ids: list()
+                Team ids to subscribe to the workspace.
+        
+        Returns
+
+            data : `dict`
```
Comment (Ashatz) [inline moncli/entities/client.py:1458]: In this case, the return object is not a dictionary but instead a list of teams.  Please update.

File: moncli/entities/client.py [hunk 5]
```diff
@@ -1273,7 +1281,337 @@ def create_workspace(self, name: str, kind: WorkspaceKind, *args, **kwargs):
...
+                utc_hours_diff : `int`
+                    The user's UTC hours difference.
+
+        Optional Arguments
+
+            api_key : `str`
+                The monday.com v2 API user key.
+       """
+        
+        workspace_data = api.add_teams_to_workspace(
```
Comment (Ashatz) [inline moncli/entities/client.py:1520]: Change this variable to _teams_data_ instead of _workspace_data_.

File: moncli/entities/client.py [hunk 6]
```diff
@@ -1273,7 +1281,337 @@ def create_workspace(self, name: str, kind: WorkspaceKind, *args, **kwargs):
...
+                workspace_id,
+                team_ids,
+                *args,
+                api_key=self.__creds.api_key_v2,
+                **kwargs)
+
+        return en.Workspace(workspace_data)
```
Comment (Ashatz) [inline moncli/entities/client.py:1527]: The return should be a list of Teams. Please use and test the following.

`return [en.Team(creds=self.__creds, **data) for data in teams_data]`

File: moncli/entities/client.py [hunk 7]
```diff
@@ -1273,7 +1281,337 @@ def create_workspace(self, name: str, kind: WorkspaceKind, *args, **kwargs):
...
+                Team ids to subscribe to the workspace.
+        
+        Returns
+
+            data : `dict`
+                A monday.com workspace in dictionary form.
+        
+        Return Fields
+
+            account : `moncli.entities.Account`
```
Comment (Ashatz) [inline moncli/entities/client.py:1547]: These are the return fields for the User object. Please use the return fields from the Team object. See get_teams and copy.

File: moncli/entities/client.py [hunk 8]
```diff
@@ -1273,7 +1281,337 @@ def create_workspace(self, name: str, kind: WorkspaceKind, *args, **kwargs):
...
+                utc_hours_diff : `int`
+                    The user's UTC hours difference.
+
+        Optional Arguments
+
+            api_key : `str`
+                The monday.com v2 API user key.
+       """
+        
+        workspace_data = api.delete_teams_from_workspace(
```
Comment (Ashatz) [inline moncli/entities/client.py:1604]: Change this variable to teams_data instead of workspace_data.

File: moncli/entities/client.py [hunk 9]
```diff
@@ -1273,7 +1281,337 @@ def create_workspace(self, name: str, kind: WorkspaceKind, *args, **kwargs):
...
+                workspace_id,
+                team_ids,
+                *args,
+                api_key=self.__creds.api_key_v2,
+                **kwargs)
+
+        return en.Workspace(workspace_data)
```
Comment (Ashatz) [inline moncli/entities/client.py:1611]: The return should be a list of Teams. Please use and test the following.

return [en.Team(creds=self.__creds, **data) for data in teams_data]

**Review by Ashatz — CHANGES_REQUESTED** [review]

File: moncli/entities/client.py (hunk 9 above)
Comment (aswath-m) [reply moncli/entities/client.py:1611]: I'm getting following error after making this change 

> File "C:\ASWATH\Github\moncli\moncli\entities\client.py", line 1502, in <listcomp>
    > return [en.Team(creds=self.__creds, **data) for data in teams_data]
     >TypeError: moncli.entities.user.Team() argument after ** must be a mapping, not str

**Comment by Ashatz** [comment]
Closing this.

**Pull request closed without merging**